index.js

```diff
@@ -4,61 +4,78 @@ const childProcess = require('child_process');
 const pify = require('pify');
 
 const bin = path.join(__dirname, 'main');
-const xprop = 'xprop -id $(xprop -root 32x \'\\t$0\' _NET_ACTIVE_WINDOW | cut -f 2)';
+const xpropActive = 'xprop -root 32x \'\\t$0\' _NET_ACTIVE_WINDOW | cut -f 2';
+const xpropDetails = 'xprop -id';
 
-function parse(stdout) {
-	if (process.platform === 'darwin') {
-		const parts = stdout.trimRight().split('\n');
+const parseMac = stdout => {
+	const parts = stdout.trimRight().split('\n');
 
-		return {
-			title: parts[0],
-			id: Number(parts[1]),
-			app: parts[2],
-			pid: Number(parts[3])
-		};
-	} else if (process.platform === 'linux') {
-		const result = {};
+	return {
+		title: parts[0],
+		id: Number(parts[1]),
+		app: parts[2],
+		pid: Number(parts[3])
+	};
+};
 
-		for (const row of stdout.trim().split('\n')) {
-			if (row.includes('=')) {
-				const parts = row.split('=');
-				result[parts[0].trim()] = parts[1].trim();
-			} else if (row.includes(':')) {
-				const parts = row.split(':');
-				result[parts[0].trim()] = parts[1].trim();
-			}
-		}
-		const windowIdProperty = 'WM_CLIENT_LEADER(WINDOW)';
-		const resultKeys = Object.keys(result);
-		const windowId = (resultKeys.indexOf(windowIdProperty) > 0 &&
-			parseInt(result[windowIdProperty].split('#').pop(), 16)) || null;
+const parseLinux = linuxData => {
+	const stdout = linuxData.stdout;
+	const activeWindowId = linuxData.activeWindowId;
 
-		return {
-			title: JSON.parse(result['_NET_WM_NAME(UTF8_STRING)']) || null,
-			id: windowId,
-			app: JSON.parse(result['WM_CLASS(STRING)'].split(',').pop()),
-			pid: parseInt(result['_NET_WM_PID(CARDINAL)'], 10)
-		};
+	const result = {};
+	for (const row of stdout.trim().split('\n')) {
+		if (row.includes('=')) {
+			const parts = row.split('=');
+			result[parts[0].trim()] = parts[1].trim();
+		} else if (row.includes(':')) {
+			const parts = row.split(':');
+			result[parts[0].trim()] = parts[1].trim();
+		}
 	}
 
-	throw new Error('macOS and Linux only');
-}
+	const windowIdProperty = 'WM_CLIENT_LEADER(WINDOW)';
+	const resultKeys = Object.keys(result);
+	const windowId = (resultKeys.indexOf(windowIdProperty) > 0 &&
+		parseInt(result[windowIdProperty].split('#').pop(), 16)) || activeWindowId;
+
+	return {
+		title: JSON.parse(result['_NET_WM_NAME(UTF8_STRING)']) || null,
+		id: windowId,
+		app: JSON.parse(result['WM_CLASS(STRING)'].split(',').pop()),
+		pid: parseInt(result['_NET_WM_PID(CARDINAL)'], 10)
+	};
+};
 
 module.exports = () => {
 	if (process.platform === 'darwin') {
-		return pify(childProcess.execFile)(bin).then(parse);
+		return pify(childProcess.execFile)(bin).then(parseMac);
 	} else if (process.platform === 'linux') {
-		return pify(childProcess.exec)(xprop).then(parse);
+		return pify(childProcess.exec)(xpropActive).then(
```

Comment:
Use `childProcess.execFile` instead. It will be faster as it calls the binary directly without launching it through a shell. You need to split the arguments as an array of arguments though.